### ISSUE TITLE
clang-format handles whitespace for "f[](x) - 7" in unexpected way

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#24729]
|  |  |
| --- | --- |
| Bugzilla Link | [24355](https://llvm.org/bz24355) |
| Version | 3.6 |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 
The incorrect behavior:
```
echo "f[](x) - 7" | clang-format
f[](x)-7
```

Similar expressions which seem to behave properly:
```
echo "f[]() - 7" | clang-format
f[]() - 7
```

```
echo "f[](1) - 7" | clang-format
f[](1) - 7
```

```
echo "f()[x] - 7" | clang-format
f()[x] - 7
```

```
echo "f()(x) - 7" | clang-format
f()(x) - 7
```

```
echo "f(x) - 7" | clang-format
f(x) - 7
```

My clang-format:
clang-format --version
Ubuntu clang-format version 3.6.0-2ubuntu1~trusty1 (tags/RELEASE_360/final) (based on LLVM 3.6.0)

I have not yet tried newer versions of clang-format to see if this has been fixed.  This seems to be incorrect behavior, but it's possible there's an option to fix this.

**Comment by mydeveloperday** [comment]
This issue seems to have been resolved  as of clang-format version 14.0.0 (https://github.com/llvm/llvm-project.git fb7bea0a5a
b96b8ba3f73653fefea881bfeae350)

$ echo "f[](x) - 7" | clang-format
f[](x) - 7

So I'm closing it

**Comment by mkurdej** [comment]
It was fixed in v3.8 already.
